commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -168,29 +183,46 @@ public void onCreate(Bundle savedInstanceState) {
             } else {
                 source = Contribution.SOURCE_EXTERNAL;
             }
-
             mimeType = intent.getType();
         }
 
         mediaUriString = mediaUri.toString();
-        Log.d("Image", "Uri: " + mediaUriString);
-
+        Log.d(TAG, "Uri: " + mediaUriString);
         //convert image Uri to file path
         FilePathConverter uriObj = new FilePathConverter(this, mediaUri);
         String filePath = uriObj.getFilePath();
 
+        //Using global singleton to get CacheController to last longer than the activity lifecycle
+        cacheObj = ((CommonsApplication)this.getApplication());
```

Comment:
I just found out that there already was an existing 'app' that did the same thing, heh. Thanks!